company.py

```diff
@@ -55,4 +55,182 @@
+
+def print_low_paid_position(departments):
+    for department in departments:
+        for employee in department['employers']:
+            if employee['salary_rub'] < 80000:
+                print(f"{employee['position']}")
+
+print_low_paid_position(departments)
+
+#6. Посчитать, сколько денег в месяц уходит на каждый отдел – и вывести вместе с названием отдела
+
+def print_department_salary(departments):
+    for department in departments:
+        department_salary = 0
+        for employee in department['employers']:
+            department_salary += employee['salary_rub']
+        print(f"{department_salary} {department['title']}")
+
+print_department_salary(departments)
+
+# 7. Вывести названия отделов с указанием минимальной зарплаты в нём.
+
+def print_min_salary(departments):
+    for department in departments:
+        min_salary = sorted([salary['salary_rub'] for salary in department['employers']])[0]
+        print(f"{department['title']} {min_salary}")
+
+print_min_salary(departments)
+
+# 8. Вывести названия отделов с указанием минимальной, средней и максимальной зарплаты в нём.
+
+def print_salary_statistics(departments):
+    for department in departments:
+        salary = [salary['salary_rub'] for salary in department['employers']]
+        print(f"{department['title']}. Min salary: {min(salary)}, average salary: {sum(salary) / len(salary)}, max salary: {max(salary)}")
+
+print_salary_statistics(departments)
+
+# Вывести среднюю зарплату по всей компании.
+
+def print_average_salary(departments):
+    company_salary = []
+    for department in departments:
+        company_salary += [salary['salary_rub'] for salary in department['employers']]
+    print(sum(company_salary) / len(company_salary))
+
+print_average_salary(departments)
+
+# 10. Вывести названия должностей, которые получают больше 90к без повторений.
+
+def print_high_paid_position(departments):
+    high_paid_position = []
+    for department in departments:
+        for employee in department['employers']:
+            if employee['salary_rub'] > 90000 and employee['position'] not in high_paid_position:
+                high_paid_position.append(employee['position'])
+    print(high_paid_position)
+
+print_high_paid_position(departments)
+
+# 11. Посчитать среднюю зарплату по каждому отделу среди девушек (их зовут Мишель, Николь, Кристина и Кейтлин).
+
+def print_average_female_salary(departments):
+    female = ['Michelle', 'Nicole', 'Christina', 'Caitlin']
+    for department in departments:
+        female_salary = [salary['salary_rub'] for salary in department['employers'] if salary['first_name'] in female]
+        print(f"{department['title']} {sum(female_salary) / len(female_salary)}")
+
+print_average_female_salary(departments)
+
+# 12. Вывести без повторений имена людей, чьи фамилии заканчиваются на гласную букву.
+
+def vowel_end_last_name(departments):
+    vowel = ['a', 'e', 'i', 'o', 'u', 'y']
```

Comment:
тоже будет работать, потому что строка тоже iterable
```suggestion
    vowel = 'aeiouy'
```